src/serve.rs

```diff
@@ -246,13 +246,10 @@ impl State {
 /// (for autoreload & HMR in the future), as well as any user-defined proxies.
 fn router(state: Arc<State>, cfg: Arc<RtcServe>) -> Result<Router> {
     // Build static file server, middleware, error handler & WS route for reloads.
-    let public_route = if state.public_url == "/" {
+    let public_route = if state.public_url.starts_with('/') {
         &state.public_url
     } else {
-        state
-            .public_url
-            .strip_suffix('/')
-            .unwrap_or(&state.public_url)
+        "/"
```

Comment:
A public URL of `abc/` would get turned into `/`? Is that correct?

Reply:
If you run `trunk serve --public-url="abc/" `, you should visit  http://127.0.0.1:8080/. I think it's correct, although it's a little strange to use a relative public-url in truck serve.

Reply:
Right, I am not sure `abc/` would be a good input value. But having to visit `/` in such a cause would also be confusing. Maybe that's the whole reason for prefixing this with a `/` in this case, so that one would get `/abc/`.

Reply:
When visiting `/abc/`, static files will be at `/abc/abc/`, I worry it will be confusing.

When people use a relative path as `public-url`, I think it's better to just refer to the location of static files, not index path. (it could be `/` or unknow)

Reply:
I see … but again, I think that putting in `abc/` and ending up with `/` is equally confusing.

Maybe it makes sense to go back to the drawing board and write down how it should work. The topic seems simple, but turns out to be more complex. I would hate to have a breaking change which still is confusing :)

Reply:
I think the key is the meaning of "public-url". When it's a relative url, I believe it's better to keep it independent of the route path. when using `trunk serve --public-url="abc/"`, you should visit `/`, but static files are still under `/abc/`.

Reply:
I have tested with `react-scripts`. It seems work differently for `build` and `start`. For build, I get `href="abc/..."` in `build/index.html`, while for `start`, it shows `http://localhost:3000/abc` and uses `href="/abc/..."`.

I'm not sure if it's a good solution. Working differently in development and production also seems strange..

Reply:
How about this:

- default to be `/` (same as react-scripts)
- never add prefix `/`
- just panic if using a public-url not staring with `/` for `serve`